### ISSUE TITLE
Scrolling in long OptionList doesn't accompany keyboard navigation by default

### ISSUE THREAD

**Issue by rodrigogiraoserrao** [Textualize/textual#4021]
If you use the keyboard to navigate inside an `OptionList` and if you move past the visible options, there is no scrolling to accompany your movement (there should).

https://github.com/Textualize/textual/assets/5621605/31e02474-b696-40d7-b300-b6348ecddd6d

<details>
<summary>App shown in video</summary>

```py
from textual.app import App, ComposeResult
from textual.widgets import OptionList
from textual.widgets.option_list import Option


class ListApp(App[None]):
    def compose(self) -> ComposeResult:
        yield OptionList(*[Option(f"This is option #{n}") for n in range(100)])


if __name__ == "__main__":
    ListApp().run()
```

</summary>

**Comment by davep** [comment]
What's the `height` for this example?

**Comment by davep** [comment]
Perhaps [*Why doesn't the DataTable scroll programmatically?*](https://textual.textualize.io/FAQ/#why-doesnt-the-datatable-scroll-programmatically) needs expanding a wee bit?

**Comment by rodrigogiraoserrao** [comment]
🤦 
Maybe `OptionList` should come with a `max-height: 100%` by default?

**Comment by TomJGooding** [comment]
> Maybe `OptionList` should come with a `max-height: 100%` by default?

I thought this fix was simple enough that I could try to help, until I remembered that  `OptionList` has padding/border. 

This might need a fix for percentage dimensions (#4037 or #4023) to land first.
